resources/pod.py

```diff
@@ -0,0 +1,158 @@
+"""
+Pod related functionalities and context info
+
+Each pod in the openshift cluster will have a corresponding pod object
+"""
+import logging
+import tempfile
+import yaml
+
+from ocs.ocp import OCP
+from ocs import defaults, kinds
+from ocsci.config import ENV_DATA
+from ocs.exceptions import CommandFailed
+
+from resources.base_resource import BaseOCSClass
+
+logger = logging.getLogger(__name__)
+
+
+class Pod(BaseOCSClass):
+    """
+    Handles per pod related context
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        kwargs:
+            Copy of ocs/defaults.py::<some pod> dictionary
+        """
+        self.pod_data = kwargs
+        super(Pod, self).__init__(
+            self.pod_data.get('apiVersion'), self.pod_data.get('kind'),
+            self.pod_data.get('metadata').get('namespace')
+        )
+
+        self.temp_yaml = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='POD_', delete=False
+        )
+        self._name = self.pod_data.get('metadata').get('name')
+        self._labels = self.get_labels()
+        self._roles = [self.pod_data.get('metadata').get('labels').get('app')]
+        self.ocp = OCP(
+            api_version=defaults.API_VERSION, kind=kinds.POD,
+            namespace=self.namespace
+        )
+        # TODO: get backend config !!
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @property
+    def roles(self):
+        return self._roles
+
+    @property
+    def labels(self):
+        return self._labels
+
+    def set_role(self, role):
```

Comment:
The method should probably be called `add_role` as it adds role to some list of roles.

Reply:
done